ironfish-cli/src/commands/ceremony.ts

```diff
@@ -0,0 +1,170 @@
+      { required: false },
+    )
+    randomness = randomness.length ? randomness : null
+
+    // Create the client and bind events
+    const client = new CeremonyClient({
+      host,
+      port,
+      logger: this.logger.withTag('ceremonyClient'),
+    })
+
+    client.onJoined.on(({ queueLocation }) => {
+      CliUx.ux.action.status = `Current position: ${queueLocation}`
+    })
+
+    client.onInitiateContribution.on(async ({ downloadLink, contributionNumber }) => {
+      CliUx.ux.action.stop()
+
+      this.log(`Starting contribution. You are contributor #${contributionNumber}`)
+
+      CliUx.ux.action.start(`Downloading the previous contribution to ${inputPath}`)
+
+      const fileHandle = await fsAsync.open(inputPath, 'w')
+
+      let response
+      try {
+        response = await axios.get(downloadLink, {
+          responseType: 'stream',
+          onDownloadProgress: (p: ProgressEvent) => {
+            this.log('loaded', p.loaded, 'total', p.total)
+          },
+        })
+      } catch (e) {
+        this.error(ErrorUtils.renderError(e))
+      }
+
+      await pipeline(response.data, fileHandle.createWriteStream())
+
+      CliUx.ux.action.stop()
+
+      CliUx.ux.action.start(`Contributing your randomness`)
+
+      localHash = await contribute(inputPath, outputPath, randomness)
+
+      CliUx.ux.action.stop()
+
+      CliUx.ux.action.start(`Waiting to upload your contribution`)
+
+      client.contributionComplete()
+    })
+
+    client.onInitiateUpload.on(async ({ uploadLink }) => {
+      CliUx.ux.action.stop()
+
+      CliUx.ux.action.start(`Uploading your contribution`)
+
+      const fileHandle = await fsAsync.open(outputPath, 'r')
+      const stat = await fsAsync.stat(outputPath)
+
+      try {
+        await axios.put(uploadLink, fileHandle.createReadStream(), {
+          maxBodyLength: 1000000000,
+          headers: {
+            'Content-Type': 'application/octet-stream',
+            'Content-Length': stat.size,
+          },
+        })
+      } catch (e) {
+        this.error(ErrorUtils.renderError(e))
+      }
+
+      CliUx.ux.action.stop()
+      client.uploadComplete()
+
+      CliUx.ux.action.start('Contribution uploaded. Waiting for server to verify')
+    })
+
+    client.onContributionVerified.on(({ hash, downloadLink, contributionNumber }) => {
+      CliUx.ux.action.stop()
+
+      if (hash !== localHash) {
+        this.error(
+          'Hashes do not match. Please contact the Iron Fish team with this error message.',
+        )
+      }
+
+      this.log(
+        `Thank you for your contribution to the Iron Fish Ceremony. You have successfully contributed at position #${contributionNumber}. The public hash of your contribution is "${hash}". You can view your contributed file at ${downloadLink}.`,
```

Comment:
another ceremony I've seen divided the hash into multiple sections. I thought it was a good way to display the hash since it's very long
```
aaaaaa aaaaaa aaaaaa
aaaaaa aaaaaa aaaaaa
aaaaaa aaaaaa aaaaaa
```

Reply:
Changed this to print out the hashes like so:
```
        95799e84 5202b5e2 e9d09f8c db981fcd
        ceec1d6e 784761e7 c4d88d5b 5ba5cb74
        dc9eda71 fa5e50c4 07957ae3 22b01fa2
        53f0c341 93b20d8b 7ee5e20f 464da9b1
```